TypeCobol/Compiler/CodeModel/SymbolTable.cs

```diff
@@ -148,62 +148,51 @@ public SymbolTable(SymbolTable enclosing, Scope current)
         public IDictionary<string, List<DataDefinition>> DataEntries =
             new Dictionary<string, List<DataDefinition>>(StringComparer.OrdinalIgnoreCase);
 
+        //Dictionary for Type data entries
+        public IDictionary<string, List<DataDefinition>> DataTypeEntries =
+            new Dictionary<string, List<DataDefinition>>(StringComparer.OrdinalIgnoreCase);
+
         internal void AddVariable([NotNull] DataDefinition symbol)
         {
             // TODO: generate a name for FILLERs and anonymous data to be referenced by in the symbol table
             if (symbol.Name == null)
             {
                 return;
             }
-            Add(DataEntries, symbol);
+            //If symbol is a part of a type add to DataTypeEntries
+            if (!symbol.IsPartOfATypeDef)
+            {
+                Add(DataEntries, symbol);
+            }
+            else
+            {
+                //Add DataDefinition for symbol in the DataTypeEntries dictionary
+                AddTypeVariableFromDataDefinition(symbol);
+            }
         }
 
-        internal void AddVariable(string name, DataDefinition data)
+        void AddTypeVariableFromDataDefinition([NotNull] DataDefinition data)
         {
-            string key = name;
-            List<DataDefinition> found;
-            bool present = DataEntries.TryGetValue(key, out found);
-            if (!present)
+            //Types are declared in the Declarations SymbolTable
+            var table = GetTableFromScope(Scope.Declarations);
+
+            if (table == null)
```

Comment:
If table is null we want to know it.
So assume it's not null so we can get the exception.